src/metadata/selectors.js

```diff
@@ -1,225 +1,181 @@
-    return getCategoryCombos(metadata)[id]
-}
-
-export const getCategoryOptionById = (metadata, id) => {
-    return getCategoryOptions(metadata)[id]
-}
-
-export const getCategoryOptionsByCoCId = (metadata, cocId) => {
-    const coc = getCategoryOptionComboById(metadata, cocId)
-    return coc.categoryOptions.map((coId) =>
-        getCategoryOptionById(metadata, coId)
-    )
-}
-
-export const getCategoryOptionComboById = (metadata, id) => {
-    return getCategoryOptionCombos(metadata)[id]
-}
-
-export const getCategoriesByCategoryComboId = (metadata, categoryComboId) => {
-    const categoryCombo = getCategoryComboById(metadata, categoryComboId)
-    const categories = categoryCombo.categories.map((id) =>
-        getCategoryById(metadata, id)
-    )
-    return categories
-}
-
-export const getCategoryOptionsByCategoryId = (metadata, categoryId) => {
-    const category = getCategoryById(metadata, categoryId)
-    const categoryOptions = category.categoryOptions.map((optId) =>
-        getCategoryOptionById(metadata, optId)
-    )
-    return categoryOptions
-}
-
-export const getCategoryOptionCombosByCategoryComboId = (
-    metadata,
-    catComboId
-) => {
-    const categoryCombo = getCategoryComboById(metadata, catComboId)
-    const categoryOptionCombos = categoryCombo?.categoryOptionCombos.map(
-        (cocId) => getCategoryOptionComboById(metadata, cocId)
-    )
-    return categoryOptionCombos
-}
+        })
+)((data, dataSetId) => dataSetId)
+
+/**
+ * This selector needs the dataSetId so it can use the selectDataElementsByDataSetId selector,
+ * which will apply the correct categoryCombo overrides.
+ */
+
+export const selectDataElementsBySection = createCachedSelector(
+    (data, _, sectionId) => selectSectionById(data, sectionId),
+    (data, dataSetId) => selectDataElementsByDataSetId(data, dataSetId),
+    (section, dataElements) =>
+        section.dataElements.map((id) =>
+            dataElements.find((de) => de.id === id)
+        )
+)((data, dataSetId, sectionId) => `${dataSetId}:${sectionId}`)
 
 /**
- * Gets the correct ordering of categoryOptionCombos within a category combination
- * @param {*} metadata
- * @param {*} catComboId
- * @returns
+ * Returns an array of objects with dataElements and their associated categoryCombos. They
+ * are grouped by categoryComboId, every time the categoryComboId changes a new group is
+ * created.
  */
-export const getSortedCategoryOptionCombosByCategoryComboId = (
-    metadata,
-    catComboId
-) => {
-    const categories = getCategoriesByCategoryComboId(metadata, catComboId)
-
-    // get options for each category, these should be in sorted order
-    const allOptionsLists = categories.map((cat) => cat.categoryOptions)
-    // compute the  combination of category-catgoryOption- this basically computes the categoryCombinations
-    // result is a matrix of categoryOptions for each column in the categoryCombo
-    const optionCombinationOrder = cartesian(allOptionsLists)
```

Comment:
Wheres the equal of this selector? The cartesian product is integral to rendering the dataValues in correct order.